HLTrigger/HLTfilters/plugins/L1EnergySumFilterT.h

```diff
@@ -0,0 +1,134 @@
+      l1tSumType_(typeOfL1TSum(iConfig.getParameter<std::string>("TypeOfSum"))),
+      minPt_(iConfig.getParameter<double>("MinPt")) {
+  scalings_ = iConfig.getParameter<edm::ParameterSet>("Scalings");
+  theScalings_ = scalings_.getParameter<std::vector<double>>("theScalings");
+}
+
+template <typename T>
+L1EnergySumFilterT<T>::~L1EnergySumFilterT() = default;
+
+template <typename T>
+trigger::TriggerObjectType L1EnergySumFilterT<T>::typeOfL1TSum(std::string const& typeOfSum) const {
+  trigger::TriggerObjectType sumEnum;
+
+  if (typeOfSum == "MET")
+    sumEnum = trigger::TriggerObjectType::TriggerL1PFMET;
+  else if (typeOfSum == "ETT")
+    sumEnum = trigger::TriggerObjectType::TriggerL1PFETT;
+  else if (typeOfSum == "HT")
+    sumEnum = trigger::TriggerObjectType::TriggerL1PFHT;
+  else if (typeOfSum == "MHT")
+    sumEnum = trigger::TriggerObjectType::TriggerL1PFMHT;
+  else {
+    throw cms::Exception("ConfigurationError")
+        << "Wrong type of energy sum: \"" << typeOfSum << "\" (valid choices are: MET, ETT, HT, MHT)";
+  }
+
+  return sumEnum;
+}
+
+template <typename T>
+void L1EnergySumFilterT<T>::fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
+  edm::ParameterSetDescription desc;
+  makeHLTFilterDescription(desc);
+  desc.add<edm::InputTag>("inputTag", edm::InputTag("L1PFEnergySums"));
+
+  edm::ParameterSetDescription descScalings;
+  descScalings.add<std::vector<double>>("theScalings", {0.0, 1.0, 0.0});
+  desc.add<edm::ParameterSetDescription>("Scalings", descScalings);
+
+  desc.add<std::string>("TypeOfSum", "HT");
+  desc.add<double>("MinPt", -1.);
+  descriptions.add(defaultModuleLabel<L1EnergySumFilterT<T>>(), desc);
+}
+
+template <typename T>
+bool L1EnergySumFilterT<T>::hltFilter(edm::Event& iEvent,
+                                      edm::EventSetup const& iSetup,
+                                      trigger::TriggerFilterObjectWithRefs& filterproduct) const {
+  // All HLT filters must create and fill an HLT filter object,
+  // recording any reconstructed physics objects satisfying (or not)
+  // this HLT filter, and place it in the Event.
+
+  // The filter object
+  if (saveTags()) {
+    filterproduct.addCollectionTag(l1tSumTag_);
+  }
+
+  auto const& l1tSums = iEvent.getHandle(l1tSumToken_);
+
+  int nSum(0);
+  for (auto iSum = l1tSums->begin(); iSum != l1tSums->end(); ++iSum) {
+    double offlinePt = 0.0;
+    // pt or sumEt?
+    if (l1tSumType_ == trigger::TriggerObjectType::TriggerL1PFMET or
+        l1tSumType_ == trigger::TriggerObjectType::TriggerL1PFMHT) {
+      offlinePt = offlineEnergySum(iSum->pt());
+    } else if (l1tSumType_ == trigger::TriggerObjectType::TriggerL1PFETT or
+               l1tSumType_ == trigger::TriggerObjectType::TriggerL1PFHT) {
+      offlinePt = offlineEnergySum(iSum->sumEt());
+    } else {
+      throw cms::Exception("Input") << "Invalid trigger::TriggerObjectType enum: " << l1tSumType_;
```

Comment:
That check/trhow should be done once in the c'tor, not inside the filter method and object loop

Reply:
I agree that the check for throwing should be done in the constructor! But I have to have an `if` there to choose the method I need to call, right? Unless there's something I am missing.

Reply:
Well, you have the throw already in ```typeOfL1TSum``` which is called in the c'tor to set ```l1tSumType_``` so then later you can assume that ```l1tSumType_``` is valid, so no further throw needed.

Reply:
Done, I take out the final `else`.